cabal-testsuite/cabal-testsuite.cabal

```diff
@@ -58,7 +58,7 @@ library
     , aeson                 ^>= 1.4.2.0 || ^>=1.5.0.0
     , async                 ^>= 2.2.1
     , attoparsec            ^>= 0.13.2.2
-    , base16-bytestring     ^>= 0.1.1.6
```

Comment:
If possible, I would recomment bumping this to just `^>=1.0.0.0` if we can get away with it. The difference in performance is massive, and the new versions are completely back-compatible for the range of GHCs cabal cares about: https://matrix.hackage.haskell.org/#/package/base16-bytestring

Reply:
I do not have a strong opinion, but if it is a question of performance and not a question of security, I'd rather avoid a cliff age (switching from `<1` to `>1`) and allow a wider range of options. There are still many packages out there, which do not support `base16-bytestring-1.0`, https://packdeps.haskellers.com/reverse/base16-bytestring

Reply:
fair.

Reply:
> I'd rather avoid a cliff age (switching from <1 to >1) and allow a wider range of options. There are still many packages out there, which do not support base16-bytestring-1.0

How is this a problem? base16-bytestring is a direct dependency (except for resolv) and cabal-install shouldn't have any reverse dependencies, should it? And both base16-bytestring-1.0 and resolv build on older ghcs too: https://matrix.hackage.haskell.org/#/package/base16-bytestring

>=1 would also save us the CPP (that we'd have to remember to remove when eventually bumping the lower bound)

Reply:
@fgaz no problem, pushed.

Reply:
@fgaz nope, it does not pass CI as configured, because `base16-bytestring-0.1.1.7` is freezed in bootstrap jsons and docker containers. Honestly, I feel that fixing them goes beyond the scope of this PR (and my role in it). I'd prefer to revert back 981f6d9, if this is fine with you.

Reply:
That's fine. Thanks for giving it a go

Reply:
Reverted 981f6d9, hopefully it is ready now.